src/state/walletOverride/reducer.ts

```diff
@@ -0,0 +1,30 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { Wallet } from 'connectors'
+
+export interface WalletOverrideState {
+  // We want the user to be able to define which wallet they want to use, even if there are multiple connected wallets via web3-react.
+  // If a user had previously connected a wallet but didn't have a wallet override set (because they connected prior to this field being added),
+  // we want to handle that case by backfilling them manually. Once we backfill, we set the backfilled field to `true`.
+  // After some period of time, our active users will have this property set so we can likely remove the backfilling logic.
+  walletOverrideBackfilled: boolean
+  walletOverride?: Wallet
```

Comment:
Rather than have `walletOverrideBackfilled`, can you just have `wallet?: Wallet | true`. If undefined, then it should be backfilled - if `true`, then no wallet has been overridden, and none need be eagerly tried. This simplifies the logic, and gets rid of a confusing parameter.